packages/core/src/sheets/typedef.ts

```diff
@@ -594,12 +594,75 @@ export interface ISingleCell {
 
 export interface IRangeWithCoord extends IPosition, IRange { }
 
-export interface ISelectionCell extends IRange, ISingleCell { }
-
+/**
+ * @deprecated use ICellWithCoord instead.
+ */
 export interface ISelectionCellWithMergeInfo extends IPosition, ISingleCell {
     mergeInfo: IRangeWithCoord; // merge cell, start and end is upper left cell
 }
 
+/**
+ * SingleCell & coordinate and mergeRange.
+ */
+// Original name: ISelectionCellWithMergeInfo
+export interface ICellWithCoord extends IPosition, ISingleCell {
+    mergeInfo: IRangeWithCoord; // merge cell, start and end is upper left cell
+
+    /**
+     * Coordinate of the single cell(actual row and column).
+     */
+    startX: number;
+    /**
+     * Coordinate of the single cell(actual row and column).
+     */
+    startY: number;
+    /**
+     * Coordinate of the single cell(actual row and column).
+     */
+    endX: number;
+    /**
+     * Coordinate of the single cell(actual row and column).
+     */
+    endY: number;
+
+    // part of singleCell
+    /**
+     * The raw row index calculated by the offsetX (Without considering merged cells, this value is simply the row index.If there are merged cells, this value refers to the cell where the mouse was clicked.)
+     */
+    actualRow: number;
+
+    /**
+     * The raw col index calculated by the offsetX (Without considering merged cells)
+     */
+    actualColumn: number;
+    /**
+     * Whether the cell is merged. But main merged cell is false.
+     */
+    isMerged: boolean;
+    /**
+     * if Merged and is main merged cell.
+     */
+    isMergedMainCell: boolean;
+}
+
+/**
+ * @deprecated  use ICellInfo instead.
+ * Range & SingleCell & isMerged.
+ * startRow: number;
+ * startColumn: number;
+ * endRow: number;
+ * endColumn: number;
+ *
+ * actualRow: number;
+ * actualColumn: number;
+ * isMerged: boolean;
+ * isMergedMainCell: boolean;
+ */
+// Bad name!! selection is a concept in sheet-ui !!
```

Comment:
Bad name? Yes. Concept in sheet-ui? Not quite if you take Node.js SDK into consideration.

Reply:
Still a bad name even node needs it. But ICellInfo isn't much better either. 
This comments is deleted, Both interfaces kept.